### ISSUE TITLE
[bug] CMakeToolchain generates invalid CMAKE_GENERATOR_TOOLSET for older toolsets+update

### ISSUE THREAD

**Issue by Nekto89** [conan-io/conan#15787]
### Environment details

* Operating System+version: Windows 11
* Compiler+version: MSVC 19.29.30151.0
* Conan version: 2.1.0
* Python version: 3.9.1

### Steps to reproduce

Install v14.29-16.11 toolset in VS2022
Create profile
```
[settings]
os=Windows
arch=x86_64
build_type=Release
compiler=msvc
compiler.version=192
compiler.update=9
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
[conf]
tools.cmake.cmaketoolchain:generator=Visual Studio 17 2022
```
Build any project that uses CMake.

conan install --req=zlib/1.3 --build="zlib/*" -pr profile

### Logs

```
conan install --req=zlib/1.3 --build="zlib/*" -pr profile

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.update=9
compiler.version=192
os=Windows
[conf]
tools.cmake.cmaketoolchain:generator=Visual Studio 17 2022

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=20
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=193
os=Windows


======== Computing dependency graph ========
Graph root
    cli
Requirements
    zlib/1.3#5c0f3a1a222eebb6bff34980bcd3e024 - Cache

======== Computing necessary packages ========
zlib/1.3: Forced build from source
Requirements
    zlib/1.3#5c0f3a1a222eebb6bff34980bcd3e024:a826e3dac739fab7a0cd5b19a85758a92034abe6 - Build

======== Installing packages ========

-------- Installing package zlib/1.3 (1 of 1) --------
zlib/1.3: Building from source
zlib/1.3: Package zlib/1.3:a826e3dac739fab7a0cd5b19a85758a92034abe6
zlib/1.3: Copying sources to build folder
zlib/1.3: Building your package in C:\Users\Nekto\.conan2\p\b\zlib9051a6e263a22\b
zlib/1.3: Calling generate()
zlib/1.3: Generators folder: C:\Users\Nekto\.conan2\p\b\zlib9051a6e263a22\b\build\generators
zlib/1.3: CMakeToolchain generated: conan_toolchain.cmake
zlib/1.3: CMakeToolchain generated: CMakePresets.json
zlib/1.3: CMakeToolchain generated: ..\..\src\CMakeUserPresets.json
zlib/1.3: Generating aggregated env files
zlib/1.3: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']
zlib/1.3: Calling build()
zlib/1.3: Apply patch (conan): separate static/shared builds, disable debug suffix, disable building examples
zlib/1.3: Running CMake.configure()
zlib/1.3: RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="C:/Users/Nekto/.conan2/p/b/zlib9051a6e263a22/b/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/Nekto/.conan2/p/b/zlib9051a6e263a22/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:\Users\Nekto\.conan2\p\b\zlib9051a6e263a22\b\src"
-- Using Conan toolchain: C:/Users/Nekto/.conan2/p/b/zlib9051a6e263a22/b/build/generators/conan_toolchain.cmake
-- Conan toolchain: CMAKE_GENERATOR_TOOLSET=version=14.29
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22631.
CMake Error at CMakeLists.txt:4 (project):
  Generator

    Visual Studio 17 2022

  given toolset and version specification

    v143,version=14.29

  contains an invalid version specification.


-- The C compiler identification is MSVC 19.37.32822.0
-- Detecting C compiler ABI info
CMake Error at C:/Users/Nekto/.conan2/p/b/zlib9051a6e263a22/b/build/CMakeFiles/CMakeScratch/TryCompile-vuuynj/CMakeLists.txt:7 (project):
  Generator

    Visual Studio 17 2022

  given toolset and version specification

    v143,version=14.29

  contains an invalid version specification.


CMake Error at C:/Program Files/CMake/share/cmake-3.25/Modules/CMakeDetermineCompilerABI.cmake:57 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.25/Modules/CMakeTestCCompiler.cmake:26 (CMAKE_DETERMINE_COMPILER_ABI)
  CMakeLists.txt:4 (project)


-- Configuring incomplete, errors occurred!
See also "C:/Users/Nekto/.conan2/p/b/zlib9051a6e263a22/b/build/CMakeFiles/CMakeOutput.log".

zlib/1.3: ERROR:
Package 'a826e3dac739fab7a0cd5b19a85758a92034abe6' build failed
zlib/1.3: WARN: Build folder C:\Users\Nekto\.conan2\p\b\zlib9051a6e263a22\b\build
ERROR: zlib/1.3: Error in build() method, line 85
        cmake.configure()
        ConanException: Error 1 while executing
```

**Comment by memsharded** [comment]
Thanks for the report @Nekto89 

This is indeed a gap, I am submitting a fix in https://github.com/conan-io/conan/pull/15789, for next Conan 2.2

**Comment by Nekto89** [comment]
> Thanks for the report @Nekto89
> 
> This is indeed a gap, I am submitting a fix in #15789, for next Conan 2.2

Thanks for the quick fix. Is there already some expected date for Conan 2.2 release? It was almost a year between Conan 2.0 and Conan 2.1.

**Comment by memsharded** [comment]
Yes, releases will be monthly now, expect this in a couple of weeks.
